### DISCUSSION TITLE
Setting cookies in Edge functions

### DISCUSSION THREAD

**Discussion by nienow** [vercel/community#3337] [Help] [answered]
### Summary

I wanted to store an access token in a cookie. It works fine in a Serverless Function, but does not work in a Edge Function. This is my first time using Edge Functions so maybe I'm missing something. Is it possible and make sense to do this in Edge Functions?

Thanks!

### Steps to Reproduce

Edge Function:
```
export const config = {
  runtime: 'edge',
};

export default function MyEdgeFunction(
  request: Request
) {
  return new Response(`Hello World`, {
    headers: {
      'Set-Cookie': 'accessToken=abc; HttpOnly;'
    }
  });
}
```

Serverless Function:
```
import type {VercelRequest, VercelResponse} from '@vercel/node';
export default function (request: VercelRequest, response: VercelResponse) {
  response.setHeader('Set-Cookie', 'accessToken=abc; HttpOnly;');
  response.status(200).send(`Hello World`);
}
```

**Comment by swarnava** [comment]
Hi,

You can get started with our sample repo here: https://vercel.com/templates/next.js/cookies

**Reply by nienow** [reply]
This example uses nextjs and appears to be setting a cookie on the client side. I am not using nextjs and need it to be set on the server side.

**Comment by nienow** [comment, marked as answer]
I figured it out. Cookies only work when using lowercase `set-cookie` header, where I was using `Set-Cookie`.
